src/components/interaction-designer/BuilderCanvas.vue

```diff
@@ -12,24 +12,63 @@
 </template>
 
 <script lang="ts">
-import { mapState } from 'vuex'
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Vue, Watch } from 'vue-property-decorator'
 import Block from '@/components/interaction-designer/Block.vue'
+import { namespace } from "vuex-class";
+import {IBlock, IFlow} from "@floip/flow-runner";
+import { IValidationStatus, debutValidationStatus } from "@/store/validation";
+import { find, isEqual, cloneDeep } from 'lodash'
 
-  @Component({
-    components: {
-      Block,
-    },
+const flowVuexNamespace = namespace('flow')
+const validationVuexNamespace = namespace('validation')
 
-    computed: {
-      ...mapState('flow', ['flows']),
-    },
-
-    mounted() {
-      this.$store.dispatch('builder/loadFlow')
-    },
-  })
+@Component({
+  components: {
+    Block,
+  },
+})
 export default class BuilderCanvas extends Vue {
+  mounted() {
+    this.$store.dispatch('builder/loadFlow')
+  }
+
+  @Watch('activeFlow', { deep: true, immediate: true })
+  async onActiveFlowChanged(newFlow: IFlow) {
+    console.debug('active flow has changed, validating ...')
+    await this.validate_flow({ flow: newFlow })
+    debutValidationStatus(this.flowValidationStatus, 'flow validation status')
+  }
+
+  @Watch('blocksOnActiveFlowForWatcher', { deep: true, immediate: true })
+  async onBlocksInActiveFlowChanged(newBlocks: IBlock[], oldBlocks: IBlock[]) {
+    if (!!newBlocks && newBlocks.length > 0) {
+      console.debug('blocks inside active flow have changed, validating ...');
+      for (let i = 0;  i < newBlocks.length; i++) {
+        const currentNewBlock = newBlocks[i]
+        const currentOldBlock = find(oldBlocks, { 'uuid': currentNewBlock.uuid })
+
+        if (isEqual(currentNewBlock, currentOldBlock)) {
+          continue; // no changes found
+        }
+
+        await this.validate_block({ block: currentNewBlock });
+        debutValidationStatus(this.blockValidationStatuses[currentNewBlock.uuid], `validation status for block ${currentNewBlock.uuid}`)
+      }
+    }
+  }
+
+  get blocksOnActiveFlowForWatcher() {
+    return cloneDeep(this.activeFlow.blocks) // needed to make comparison between new & old values on watcher
```

Comment:
We can improve performance here to not watch `vendor_metadata` changes (eg: delete it before cloning). I tried but it didn't work yet, it may require extra debug & tests. I'd suggest we move the all watching improvements in Beta version.

@bzabos , any thoughts ?

Reply:
I think this would speed up cloning, but I'm curious whether or not it would speed up validation if there are not any rules to validate against that data? I'd be slightly worried that a future/vendor block type could attempt to validate a property within `vendor_metadata` and be prevented from doing so! Do you think that's a fair concern?

Reply:
Hum, you make a point here. Let's not delete the  `vendor_metadata` (if we decide to keep this getters, see later below my explanation of why I put `cloneDeep`)